### DISCUSSION TITLE
QRImageWithLogo more than 1 icon

### DISCUSSION THREAD

**Discussion by uchup** [chillerlan/php-qrcode#262] [General]
can add more than 1 icon to the qr code in QRImageWithLogo?

**Comment by codemasher** [comment]
You can add as many icons you want, you'd just have to hand over the logo file sources properly (or hardcode them).
The example merely serves as a suggestion on *how* to do it.

**Reply by uchup** [reply]
![Screenshot from 2024-04-26 08-52-29](https://github.com/chillerlan/php-qrcode/assets/4725602/53b6839e-2b18-420c-b77c-dbd2a5b96827)
can add an image like this? because QRImageWithLogo only provides 1 place in the middle, and in the dump input
How can I add more than 1 image? like image 2

**Reply by codemasher** [reply]
Basically you duplicate this part and modify the settings for the other image:
https://github.com/chillerlan/php-qrcode/blob/0a897ca7235757a6e59e0949157daa5e8f7c680b/examples/imageWithLogo.php#L46-L60

**Reply by uchup** [reply]
![Screenshot from 2024-04-26 13-42-40](https://github.com/chillerlan/php-qrcode/assets/4725602/50aa8d8a-8e52-4263-994d-90edbce75866)
I've replayed it, but the 2nd image covers the QR code
unlike image 1 where the qrcode is skipped
Will the qrcode remain valid if it is overwritten by the 2nd image?

**Reply by codemasher** [reply]
If that image is transparent enough, maybe? You better test it thoroughly before deploying.
As an alternative i can recommend this example: https://github.com/chillerlan/php-qrcode/blob/v5.0.x/examples/imagickImageAsBackground.php
